### PR TITLE
add method for making the polling timeout configurable

### PR DESCRIPTION
**Pull request by LKDevelopment**
This PR aims to make it possible to reduce the number of action pollings.

### CODE REVIEW

File: hcloud/action.go [hunk 1]
```diff
@@ -141,14 +141,21 @@ func (c *ActionClient) All(ctx context.Context) ([]*Action, error) {
 
 // WatchProgress watches the actions progress until it completes with success or error.
 func (c *ActionClient) WatchProgress(ctx context.Context, action *Action) (<-chan int, <-chan error) {
+
+	return c.WatchProgressMS(ctx, action, 500)
+}
+
+// WatchProgressMS watches the actions progress until it completes with success or error. Makes it possible to configuration the polling timeout.
+func (c *ActionClient) WatchProgressMS(ctx context.Context, action *Action, timeoutMs int) (<-chan int, <-chan error) {
```
Comment (thcyron) [inline hcloud/action.go]: Third parameter should be `timeout time.Duration`. Then I would suggest to rename the method to `WatchProgressTimeout`.

**Comment by thcyron** [comment]
Please consult the [time.Duration docs](https://godoc.org/time#Duration).

File: hcloud/action_test.go [hunk 2]
```diff
@@ -184,7 +184,7 @@ func TestActionClientWatchProgress(t *testing.T) {
 	}
 
 	ctx := context.Background()
-	progressCh, errCh := env.Client.Action.WatchProgress(ctx, action)
```
[the remark this answers is not in the document]
Comment (LKDevelopment) [reply hcloud/action_test.go]: The idea was to test the correct implementation all the time, not only the wrapper around it

File: hcloud/action.go [hunk 3]
```diff
@@ -141,14 +141,21 @@ func (c *ActionClient) All(ctx context.Context) ([]*Action, error) {
 
 // WatchProgress watches the actions progress until it completes with success or error.
 func (c *ActionClient) WatchProgress(ctx context.Context, action *Action) (<-chan int, <-chan error) {
+
+	return c.WatchProgressTimeout(ctx, action, 500 * time.Millisecond)
+}
+
+// WatchProgressTimeout watches the actions progress until it completes with success or error. Makes it possible to configuration the polling timeout.
+func (c *ActionClient) WatchProgressTimeout(ctx context.Context, action *Action, timeout time.Duration) (<-chan int, <-chan error) {
```
[the remark this answers is not in the document]
Comment (LKDevelopment) [reply hcloud/action.go]: Yeah interval sounds better, i will change it.

**Pull request merged**